Don't reject URIs with encoded '/' with servlet 6

idlewis opened the pull request:
Fix for issue #23570

idlewis committed:
Servlet 6 changes in jmx rest connector

Add property in web.xml so that the servlet container doesn't
reject URIs which contain URL encoded '/'

Add code to repair the damage if the container collapses '//'
down to '/' in the URI, as a URI segment can contain an mbean
name, and an mbean name could contain '//'.

idlewis commented:
#build

idlewis commented:
#build

idlewis commented:
#build

idlewis commented:
#run-libby-bot

idlewis commented:
https://wasrtc.hursley.ibm.com:9443/jazz/web/projects/WS-CD#action=com.ibm.team.build.viewResult&id=_JhKK8JsDEe25i_tSaDMOdg
build with full fat buckets for jax-ws and jmx rest connector client

leochr commented on dev/com.ibm.ws.jmx.connector.server.rest/src/com/ibm/ws/jmx/connector/server/rest/handlers/MBeanOperationHandler.java:
```diff
@@ -1,5 +1,5 @@
 /*******************************************************************************
- * Copyright (c) 2016 IBM Corporation and others.
+ * Copyright (c) 2016,2023 IBM Corporation and others.
  * All rights reserved. This program and the accompanying materials
```
@idlewis wondering whether the repairSlashes method be called within the _deleteMBean_ method? 
https://github.com/OpenLiberty/open-liberty/blob/e9ab3f3e1b5a82af90b47b4730ca3e9b5d7f1280/dev/com.ibm.ws.jmx.connector.server.rest/src/com/ibm/ws/jmx/connector/server/rest/handlers/MBeanHandler.java#L173

idlewis replied:
Yes, I must have missed that one :-( . I'll have to add this and rerun the build.

leochr commented on dev/com.ibm.ws.jmx.connector.server.rest/src/com/ibm/ws/jmx/connector/server/rest/handlers/MBeanAttributeRouterHandler.java:
```diff
@@ -1,5 +1,5 @@
 /*******************************************************************************
```
The objectName is not stored for other methods like https://github.com/OpenLiberty/open-liberty/blob/e9ab3f3e1b5a82af90b47b4730ca3e9b5d7f1280/dev/com.ibm.ws.jmx.connector.server.rest/src/com/ibm/ws/jmx/connector/server/rest/handlers/MBeanAttributeRouterHandler.java#L144

Just want to make sure that they don't need any changes.

idlewis replied:
I couldn't see any reason to make any changes here. There don't seem to be any side effects of calling getRequiredParam. My guess is that the call is there to validate that the parameter exists, rather than to actually retrieve the value, and this will continue to work under servlet 6

leochr reviewed (dismissed):
@idlewis Added a couple of questions/clarifications. Looks good otherwise. Thank you

idlewis committed:
Add missed change to delete method in MBeanHandler

idlewis commented:
https://wasrtc.hursley.ibm.com:9443/jazz/web/projects/WS-CD#action=com.ibm.team.build.viewResult&id=_UxHD4JvBEe25i_tSaDMOdg
New build after addressing review comments

leochr reviewed (approved):
@idlewis Thanks for the update. Looks good.

idlewis commented:
#run-libby-bot

idlewis commented:
Most recent build only had unrelated failures
